### ISSUE TITLE
Escape sequences not handled properly

### ISSUE THREAD

**Issue by thefossguy** [NixOS/nixfmt#189]
### Description

It appears that `nixfmt` does not preserve backslashes. I have the following two lines in my `darwin.nix`:

https://github.com/thefossguy/prathams-nixos/blob/6acda3f9c27516b81a29d718ea1cb3b1025f86b6/nixos-configuration/home-manager/darwin.nix#L56-L57

And this got mangled like so:

```diff
-      "^\Uf702" = "moveWordLeft:"; # Ctrl-<Left>
-      "^\Uf703" = "moveWordRight:"; # Ctrl-<Right>
+      "^Uf702" = "moveWordLeft:"; # Ctrl-<Left>
+      "^Uf703" = "moveWordRight:"; # Ctrl-<Right>
```

**Comment by piegamesde** [comment]
Nix repl does the same btw

```
nix-repl> "^\Uf702"
"^Uf702"
```

**Comment by thefossguy** [comment]
I don't even remember why I have that key _specific_ string in the key binding but it looks like Nix is actually writing the "incorrect" string to the target file.

https://github.com/nix-community/home-manager/blob/e84811035d7c8ec79ed6c687a97e19e2a22123c1/modules/targets/darwin/keybindings.nix#L41-L42

Contents of `~/Library/KeyBindings/DefaultKeyBinding.dict`:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>^Uf702</key>
	<string>moveWordLeft:</string>
	<key>^Uf703</key>
	<string>moveWordRight:</string>
</dict>
</plist>
```

And it works. So not an issue at all. Sorry for the trouble :)

**Comment by infinisil** [comment]
> Not sure if nixfmt should preserve the input regardless or not. On the one hand I want it to treat strings as opaque and never change them, on the other hand normalizing this here has the potential of uncovering very subtle bugs.

I think we should preserve the input regardless, because it's not nixfmt's job to fix these kind of problems. Instead it should be Nix itself that does that.

And we do have that [in the standard](https://github.com/NixOS/nixfmt/blob/master/standard.md#strings):

> The non-interpolated string parts must be preserved from the input.

And by preserving this we can also simplify the code because we don't need to think as much about how Nix parses strings.

If you don't mind, let's leave this open for now :)

**Comment by infinisil** [comment]
We very briefly discussed this in todays team meeting:

  - Not entirely sure how this ends up happening at all
  - Must be somewhere in the parser
  - @infinisil: The strings shouldn't be touched at all, would be good to figure this out

**Comment by piegamesde** [comment]
Turns out this was already fixed months before this bug was opened in a75658d892846bd235ed48ee10ca2754d1e0b906. I checked and indented strings are behaving correctly as well, thanks to a fix by Infinisil in ea74366f0170ff30daa5ef7c3fcf9049c425217e
